src/confd.service

```diff
@@ -5,7 +5,11 @@ Before=rsyslog.service
 [Service]
 Type=oneshot
 EnvironmentFile=/etc/docker.env
-ExecStart=-/usr/local/bin/confd -onetime -confdir=/etc/confd -node=http://172.17.42.1:4001
+ExecStart=-/bin/bash -c \
+'if [[ $CONFD_BACKEND = "ENV" ]]; \
+then /usr/local/bin/confd -onetime -confdir=/usr/src/app/config/confd_env_backend -backend env; \
```

Comment:
Why are we targeting a different place for the config?

Reply:
/etc/confd is a [symlink to /usr/src/app/config/confd](https://github.com/resin-io/resin-base/blob/7746bab596b8cd767981be4d1b694e0f3c661f0c/Dockerfile#L53).
During the transition from etcd to envvars, would we like to keep both options, with the ability to switch with CONFD_BACKEND envvar?

Reply:
I guess we ll start duplicating confd configuration, making adjustments for envvars (in toml and templates), and keeping both usr/src/app/config/confd and usr/src/app/config/confd_env_backend during the transition period.

Reply:
Is it not possible to have cloudformation provide both env vars and etcd, and then each service can just switch over immediately rather than having duplicated config files in every service which will cause headaches when trying to change them?

Reply:
That would be a great and clear approach (cc @brownjohnf )!

Currently, envvar confd back-end support was focused on migration to kubernetes. We can (rather soon) start moving one component at a time to the k8s cluster, and enabling confd envvar back-end in one step (using a script for verifying the files produced are in sync in the process). The current plan is indeed more complex when changes are required, but seems simpler and requires fewer steps/deployments in the process. Should we revise that (cc @petrosagg  @jviotti) ?

Reply:
Can you elaborate on your idea, Page? The fleet services for the components come from resin-containers on the manager. Are you saying cloudformation should write env files on all the hosts which are then read by the fleet services? We're also moving application configuration out of cf and into the resin-io-secrets repo, so that adds an additional complication.